pkg/kp/pcstore/consul_store.go

```diff
@@ -151,3 +162,72 @@ func kvpToPC(pair *api.KVPair) (fields.PodCluster, error) {
 
 	return pc, nil
 }
+
+// Watch implements a watch for the Pod cluster at _id_
+// It will return a blocking channel on which the client can read
+// WatchedPodCluster objects. The goroutine maintaining the watch will block on
+// writing to this channel so it's up to the caller to read it with haste.
+// One blocking watch will not affect any of the other watches.
+func (s *consulStore) Watch(id fields.ID, quit <-chan struct{}) <-chan WatchedPodCluster {
+	inCh := make(chan api.KVPairs)
+	outCh := make(chan WatchedPodCluster)
+	errChan := make(chan error, 1)
+
+	go consulutil.WatchPrefix(podClusterTree, s, inCh, quit, errChan)
+
+	go func() {
+		for {
+			select {
+			case <-quit:
+				return
+			case err := <-errChan:
+				s.logger.WithError(err).Errorf("WatchPrefix returned error, recovered.")
+			}
+		}
+	}()
+
+	go func() {
+		for kvp := range inCh {
+			if kvp == nil {
+				// nothing to do
+				continue
+			}
+
+			pcs, err := kvpsToPC(kvp)
+			if err != nil {
+				select {
+				case <-quit:
+					return
+				case outCh <- WatchedPodCluster{PodCluster: nil, Err: err}:
+				}
+			}
+
+			for _, pc := range pcs {
+				select {
+				case outCh <- WatchedPodCluster{PodCluster: &pc, Err: nil}:
+				case <-quit:
+					return
+				}
+			}
+		}
+	}()
+
+	return outCh
+}
+
+func (s *consulStore) List(prefix string, opts *api.QueryOptions) (api.KVPairs, *api.QueryMeta, error) {
```

Comment:
This is necessary to implement the `ConsulLister` interface as required by WatchPrefix. I could instead pass `s.kv` into `WatchPrefix` which might work.